### ISSUE TITLE
Please help getting started

### ISSUE THREAD

**Issue by bukephalos** [bmeneg/coc-perl#18]
I installed everything (Perl::LanguageServer, Node.js, coc-nvim and coc-perl, all within Linux Mint 21.1)
Perl::LanguageServer works with VScode. As a short test I created a very short Perl file, reformated it, added a syntax error which was reported, so the Server works as expected.

Within vim I tried the same after installing coc-perl. Since I did not want to rely on a key binding I did the first test with a command: :call CocActionAsync('format') -- no effect (Perl::Tidy is installed)

:CocInfo gives this:
```
## versions
  
  vim version: VIM - Vi IMproved 8.2 8024919
  node version: v19.7.0
  coc.nvim version: 0.0.82-b7375d5f 2023-01-30 05:09:03 +0800
  coc.nvim directory: /home/myuser/.vim/plugged/coc.nvim
  term: dumb
  platform: linux
  
  ## Log of coc.nvim
  
  2023-03-03T18:10:47.527 INFO (pid:2723) [extension:coc-perl] - extension "perl" is now active
  2023-03-03T18:10:47.527 INFO (pid:2723) [extension:coc-perl] - cmd: perl args: -MPerl::LanguageServer -e Perl::LanguageServer::run -- --p  ort 13604 --log-level 1 --log-file  --version 2.3.0
  2023-03-03T18:10:47.532 INFO (pid:2723) [plugin] - coc.nvim initialized with node: v19.7.0 after 186
  2023-03-03T18:10:47.541 INFO (pid:2723) [language-client-index] - Language server "perl" started with 2738
  2023-03-03T18:11:08.724 INFO (pid:2723) [attach] - receive notification: format []
  2023-03-03T18:11:08.727 ERROR (pid:2723) [attach] - Error on notification format Error: Format provider not found for buffer: 1
      at ob.documentFormat (/home/myuser/.vim/plugged/coc.nvim/build/index.js:261:28087)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async ob.formatCurrentBuffer (/home/myuser/.vim/plugged/coc.nvim/build/index.js:261:27847)
      at async _b.cocAction (/home/myuser/.vim/plugged/coc.nvim/build/index.js:281:46242)
      at async EventEmitter.<anonymous> (/home/myuser/.vim/plugged/coc.nvim/build/index.js:281:47836)
  2023-03-03T18:11:09.026 INFO (pid:2723) [attach] - receive notification: highlight []
[ and more highlight lines ... ]
```
Then I also changed the code to trigger a syntax error warning but nothing.
I also tried to jump from a function call to the definition but nothing happens. (I tried the .vimrc example from the coc.nvim README and then "gd")

So I seem to have a running extension that does nothing. What am I missing?

**Comment by bmeneg** [comment]
@bukephalos, I'm really sorry for the "after months" reply.
I just updated `coc-perl` with new versions of both `coc-nvim` and `Perl::LanguageServer`, which were not properly sync'ed in previous version and might have been the cause of your error in VIM. Could you recheck it and tell me the results?

Also, did you installed `coc-perl` through `:CocInstall coc-perl` or vim-plug?

**Comment by bukephalos** [comment]
Thanks for looking into it.
I guess did something wrong (I am not very familiar with coc). I tried to get a clean install: moved ~/.vim, .vimrc and ~/.config/coc away, reinstalled coc.nvim following the native instructions to install coc.nvim using git, Did a :CocInstall coc-perl and got this error message in CocInfo:

023-07-07T11:54:22.899 ERROR (pid:1996) [extensions-manager] - Error on activate coc-perl RangeError [ERR_SOCKET_BAD_PORT]: options.port should be >= 0 and < 65536. Received type string ('136030').

Where does this strange port come from?

**Comment by bukephalos** [comment]
Addition: In vscode Extension settings I found a setting "debug port" set to 13603. Looks like coc-perl adds a 0 to it?

**Comment by bmeneg** [comment]
> Addition: In vscode Extension settings I found a setting "debug port" set to 13603. Looks like coc-perl adds a 0 to it?

That's on me! Sorry about that. I'm preparing a new minor release to get it fixed.

**Comment by bmeneg** [comment]
@bukephalos the issue with debug port is fixed now. Feel free to update `coc-perl` (v1.2.1).

**Comment by bukephalos** [comment]
Looks good! Thanks for the fix.

**Comment by bukephalos** [comment]
Sorry to bother you again but after some more testing it seems to stop working -- this time without any error message.
What I did:
- Install your most recent version. When the install was done everything looked very good. Errors were marked and the corresponding Perl error messages shown below.
- Then I closed vim and opened the same file again. Now the errors were still marked but the messages are no longer shown.
- After more open/close cycles or creating a new file also with a syntax error nothing is shown. Just the Perl code but no error indicator. CocInfo gives no error and lookes as if the Language Server was started/connected. But as I said, no noticable effect.
- I did a reboot and was back to the second state: error indicators (two ancle brackets to the left of the lines with syntax errors) but no messages about the type of error.
- I fixed one error (the file contained several syntax errors) and all error indicators disappeared. Very strange.

Do you have any idea what could have happened?

**Comment by bmeneg** [comment]
@bukephalos, I'm glad that the original problem was fixed :)

But for the next issue, that's the exact behavior I saw when I was trying to figure out the issue reported on #19. Unfortunately that's a bug on language server, to which I'm already preparing the fix. I'll keep digging and reporting any updates on issue #19 if you don't mind, but I'll copy the points you mentioned because it was the perfect description for the bug.
